bundles/org.openhab.binding.mqtt.generic/src/main/java/org/openhab/binding/mqtt/generic/tools/DelayedBatchProcessing.java

```diff
@@ -56,18 +56,19 @@ public DelayedBatchProcessing(int delay, Consumer<List<T>> consumer, ScheduledEx
     }
 
     /**
-     * Add new object to the batch process list. If the list was empty, the delay timer
-     * is armed and all successive objects are accumulated from here on.
+     * Add new object to the batch process list. Every time a new object is received,
+     * the delay timer is rescheduled.
      *
      * @param t An object
      */
     @Override
-    public void accept(T t) {
+    synchronized public void accept(T t) {
```

Comment:
Do you really expect this method to be called concurrently?

Reply:
Yes, sir, I have seen more than on scheduler been created with a single Homie device and I can't synchronize on the ```future``` field because it is nullable :wink: